src/main.rs

```diff
@@ -435,6 +451,52 @@ impl HabitCtl {
         false
     }
 
+        fn habit_skipified(&self, habit: &Habit, date: &Date<Local>) -> bool {
+        if habit.every_days < 1 {
+            return false;
+        }
+
+        let from = date
+            .checked_sub_signed(chrono::Duration::days(habit.every_days - 1))
+            .unwrap();
+        let mut current = from;
+        while current <= *date {
+            if let Some(entry) = self.get_entry(&current, &habit.name) {
+                if entry.value == "s" {
+                    return true;
+                }
+            }
+            current = current
+                .checked_add_signed(chrono::Duration::days(1))
+                .unwrap();
+        }
+        false
+    }
```

Comment:
If I'm correct, this iterates over all days from `date` (the argument) minus `habit.every_days`, and checks if entries for any of those days include a skipped log entry, right?

Reply:
Pretty much. This is exactly the same code as the code for satsified except for the fact it detects a skip instead of done. This could probably be refactored with the satified function to have a passed Done or Skip but it just seemed clearer to leave it this way.